cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -72,473 +40,46 @@ type Server struct {
-	}
-	detail.DisplayName = displayName
-
-	// Carvel uses base64-encoded SVG data for IconSVGBase64, whereas we need
-	// a url, so convert to a data-url.
-	iconSVGBase64, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "iconSVGBase64")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "required field spec.iconSVGBase64 not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
-	}
-	if found && iconSVGBase64 != "" {
-		detail.IconUrl = fmt.Sprintf("data:image/svg+xml;base64,%s", iconSVGBase64)
-	}
-
-	shortDescription, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "shortDescription")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "required field spec.shortDescription not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
-	}
-	if found {
-		detail.ShortDescription = shortDescription
-	}
-
-	categories, found, err := unstructured.NestedStringSlice(pkgMeta.Object, "spec", "categories")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "required field spec.categories not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
-	}
-	if found {
-		detail.Categories = categories
-	}
-
-	detail.Version = &corev1.PackageAppVersion{
-		PkgVersion: version,
-	}
-
-	longDescription, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "longDescription")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "required field spec.longDescription not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
-	}
-	if found {
-		detail.LongDescription = longDescription
-	}
-	detail.Readme = longDescription
-
-	supportDescription, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "supportDescription")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "required field spec.supportDescription not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
-	}
-	if found {
-		detail.Readme = detail.Readme + "\n\n" + supportDescription
-	}
-
-	maintainersUnstructured, found, err := unstructured.NestedSlice(pkgMeta.Object, "spec", "maintainers")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "required field spec.maintainers not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
-	}
-	if found {
-		maintainers := []*corev1.Maintainer{}
-		for _, m := range maintainersUnstructured {
-			maintainerMapUnstructured, ok := m.(map[string]interface{})
-			if !ok {
-				continue
+				return nil, nil, status.Errorf(codes.FailedPrecondition, fmt.Sprintf("unable to get dynamic client : %v", err))
 			}
-			maintainerMap, found, err := unstructured.NestedStringMap(maintainerMapUnstructured)
-
-			if found && err == nil {
-				maintainers = append(maintainers, &corev1.Maintainer{
-					Name:  maintainerMap["name"],
-					Email: maintainerMap["email"],
-				})
```

Comment:
Just a note for next time: You can usually avoid massive diffs of moved code like this by just `git mv`ing the file which will change less and create a new one in its place once moved. Git will then (usually) track it as such so it's much easier to see if something changed in the moved code etc.

Reply:
Well, this particular diff I didn't know how to start with. I mean, I opted to remove the old and then add it back in the proper PR. Perhaps moving the old code to an auxiliary file and then splitting pieces of code out from there could have been another option. I agree my way is too verbose and adds unnecessary code-review complexity